geospace-http/src/main/scala/com/socrata/geospace/http/GeospaceParams.scala

```diff
@@ -0,0 +1,44 @@
+package com.socrata.geospace.http
+
+import org.scalatra.{BadRequest, ScalatraBase}
+import scala.util.Try
+
+trait GeospaceParams { this: ScalatraBase =>
+  //scalastyle:off
+  private def invalidParamError(paramName: String) =
+    halt(BadRequest(s"Invalid $paramName param provided in the request"))
+  private def missingParamError(paramName: String) =
+    halt(BadRequest(s"No $paramName param provided in the request"))
+  //scalastyle:on
+
+  def optionalInput[T](name: String,
+                       default: Option[T],
+                       extract: String => Option[String],
+                       convert: String => T): Option[T] = {
+    def tryConvert(raw: String): T = Try(convert(raw)).getOrElse(invalidParamError(name))
+    params.get(name).map(tryConvert).orElse(default)
+  }
+
+  def mandatoryInput[T](name: String,
+                        default: Option[T],
+                        extract: String => Option[String],
+                        convert: String => T): T =
+    optionalInput(name, default, extract, convert).getOrElse(missingParamError(name))
+
+  def queryParam(name: String): Option[String] = params.get(name)
+  def header(name: String): Option[String]     = request.headers.get(name)
+
+  def mandatoryQueryParam(name: String, default: Option[String] = None): String =
+    mandatoryInput(name, default, queryParam, _.toString)
+  def mandatoryHeader(name: String, default: Option[String] = None): String =
+    mandatoryInput(name, default, header, _.toString)
+
+  def resourceName: String = mandatoryQueryParam("resourceName")
+  def friendlyName: String = mandatoryQueryParam("friendlyName")
+  def forceLonLat: Boolean = mandatoryInput("forceLonLat", Some(false), queryParam, _.toBoolean)
+  def bypassValidation: Boolean = mandatoryInput("bypassValidation", Some(false), queryParam, _.toBoolean)
+
+  def authToken: String = mandatoryHeader("Authorization")
```

Comment:
If we make it a lazy val, it will be evaluated once for the lifetime of the servlet. We want to reevaluate it every time a new request comes in because each request has different parameters and headers.

-----Original Message-----
From: "AEscobarCruz" <notifications@github.com>
Sent: ‎3/‎10/‎2015 12:09
To: "socrata/geospace" <geospace@noreply.github.com>
Cc: "urmilan" <urmila.nadkarni@socrata.com>
Subject: Re: [geospace] Replace socrata-cloudbees-sbt plugin with newsocrata-sbt-plugins (#44)

In geospace-http/src/main/scala/com/socrata/geospace/http/GeospaceParams.scala:
> +    optionalInput(name, default, extract, convert).getOrElse(missingParamError(name))
> +
> +  def queryParam(name: String): Option[String] = params.get(name)
> +  def header(name: String): Option[String]     = request.headers.get(name)
> +
> +  def mandatoryQueryParam(name: String, default: Option[String] = None): String =
> +    mandatoryInput(name, default, queryParam, _.toString)
> +  def mandatoryHeader(name: String, default: Option[String] = None): String =
> +    mandatoryInput(name, default, header, _.toString)
> +
> +  def resourceName: String = mandatoryQueryParam("resourceName")
> +  def friendlyName: String = mandatoryQueryParam("friendlyName")
> +  def forceLonLat: Boolean = mandatoryInput("forceLonLat", Some(false), queryParam, _.toBoolean)
> +  def bypassValidation: Boolean = mandatoryInput("bypassValidation", Some(false), queryParam, _.toBoolean)
> +
> +  def authToken: String = mandatoryHeader("Authorization")
Not sure, but what about making these (resourceName, friendlyName, etc..) lazy vals instead? 
—
Reply to this email directly or view it on GitHub.